examples_tests/13.BRDF_Explorer/BRDFExplorerApp.cpp

```diff
@@ -374,19 +388,33 @@ R"(
 		// there are better identities to get all of these
 		const vec3 V = normalize(uEyePos - WorldPos);
 		const vec3 L = normalize(relLightPos);
+        const vec3 H = normalize(V+L);
 
 		float NdotV = dot(N, V);
         float LdotV = dot(L, V);
 
         // dots with H identities taken from Earl Hammon's PBR Diffuse Lighting GDC17 lecture
+        /*
         const float LplusV_lenSq = 2.0 + 2.0*LdotV;
         const float LplusV_rcpLen = inversesqrt(LplusV_lenSq);
         const float NdotH = max((NdotL + NdotV) * LplusV_rcpLen, 0.0);
         const float VdotH = max(LplusV_rcpLen + LplusV_rcpLen*LdotV, 0.0);
+        */
+        const float NdotH = max(dot(N, H), 0.0);
+        const float VdotH = max(dot(V, H), 0.0);
```

Comment:
no `max` necessary, its impossible to have a view vector at an obtuse angle to the half vector!

Assuming normalized V and L

H = (V+L)/Positive_Divisor
VdotH = (1+VdotL)/Positive_Divisor
VdotL >= -1.0